ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,391 @@
+        #       actual remote sibling in wording
+        ds_siblings = [r['name'] for r in ds.siblings(result_renderer=None)]
+        if not force and name in ds_siblings:
+            yield get_status_dict(
+                status='error',
+                message="a sibling '{}' is already configured. Use --force to overwrite it.".format(name),
+                **res_kwargs,
+            )
+            return
+
+        if not no_ria_remote and not force and ria_remote in ds_siblings:
+            yield get_status_dict(
+                status='error',
+                message="a storage-sibling '{}' is already configured. Use --force to overwrite it.".format(ria_remote),
+                **res_kwargs,
+            )
+            return
+
+        # parse target URL
+        # Note: For python API we should be able to deal with RI and its subclasses. However, currently quite a dance
+        # needed (see below)
+        src_url_ri = RI(url) if not isinstance(url, RI) else url
+        if src_url_ri.fragment:
+            # TODO: This is still somewhat fragile. If the given dataset id in fragment matches, it's actually fine.
+            #       But then: What if additional things are given (like @somebranch)?
+            #       However, ideally we expect no fragment at all.
+            lgr.warning("Ignoring unexpected URL fragment '%s'." % src_url_ri.fragment)
+
+        # check special remote config:
+        base_path = src_url_ri.path
+        #if not base_path:
+        #    # TODO: consider annexconfig the same way the special remote does (in-dataset special remote config)
+        #    base_path = ds.config.get("annex.ria-remote.{}.base-path".format(ria_sibling), None)
+        if not no_ria_remote and not ds.config.get("annex.ria-remote.{}.base-path".format(ria_remote), None):
+            yield get_status_dict(
+                status='impossible',
+                message="Missing required configuration 'annex.ria-remote.{}.base-path'".format(ria_remote),
+                **res_kwargs,
+            )
+            return
+
+        base_path = Path(base_path)
+
+        # append dataset id to url and use magic from clone-helper:
+        # TODO: This dance in URL parsing should be centralized in datalad-core
+        src_url_ri.fragment = ds.id
+        # Note: Attention, decode_source_spec changes the passed RI. That's why we read base_path before
+        # (as the RI's .path will be appended by the actual repo path afterwards):
+        target_props = decode_source_spec(src_url_ri, cfg=ds.config)
+        if target_props['type'] != 'ria':
+            raise ValueError("Not a valid RIA URL: %s. Expected: 'ria+[http|ssh|file|...]://[base-path]'")
+        # TODO: What sanity checks do we need?
+        #       - ria+ scheme really required? If not, ds.id in fragment doesn't do anything. Can still be valid? How
+        #         then to distinguish base-path from repo-path?
+
+        ssh_host = src_url_ri.hostname
```

Comment:
FTR: If https://github.com/datalad/datalad/pull/4064 ever lands, URL rewriting should be applied before a final hostname (if there still is any) is determined.